src/display/annotation_storage.js

```diff
@@ -49,6 +53,9 @@ class AnnotationStorage {
    * @param {Object} value
    */
   setValue(key, value) {
+    if (this._storage.get(key) !== value) {
+      this.setModified();
```

Comment:
The modification logic should get a unit test. The unit tests for the annotation storage are in https://github.com/mozilla/pdf.js/blob/master/test/unit/annotation_storage_spec.js. I would suggest to add a new one that adds the `onSetModified` and `onResetModified` callbacks and asserts that they are called when `setModified` and `resetModified` are called, and when `setValue` is used, basically covering all lines added here.